bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java

```diff
@@ -0,0 +1,191 @@
+    }
+  }
+
+  @Test
+  public void testGenerateWorkload() {
+    BigtableWorkloadOptions options = PipelineOptionsFactory.create()
+        .as(BigtableWorkloadOptions.class);
+    options.setBigtableInstanceId(instanceId);
+    options.setBigtableTableId(TABLE_ID);
+    options.setRegion(REGION_ID);
+
+    Pipeline p = Pipeline.create(options);
+
+    CloudBigtableTableConfiguration bigtableTableConfig =
+        new CloudBigtableTableConfiguration.Builder()
+            .withProjectId(options.getProject())
+            .withInstanceId(options.getBigtableInstanceId())
+            .withTableId(options.getBigtableTableId())
+            .build();
+
+    // Initiates a new pipeline every second
+    p.apply(Create.of(1L))
+        .apply(ParDo.of(new ReadFromTableFn(bigtableTableConfig)));
+    p.run().waitUntilFinish();
+
+    String output = bout.toString();
+    assertThat(output.contains("Connected to table"));
+  }
+
+  @Test
+  public void testPipeline() throws IOException, InterruptedException {
+    String workloadJobName = "bigtable-workload-generator-test-" + new Date().getTime();
+    final int WAIT_DURATION = 10 * 60 * 1000;
+    int rate = 1000;
+
+    BigtableWorkloadOptions options = PipelineOptionsFactory.create()
+        .as(BigtableWorkloadOptions.class);
+    options.setBigtableInstanceId(instanceId);
+    options.setBigtableTableId(TABLE_ID);
+    options.setWorkloadRate(rate);
+    options.setRegion(REGION_ID);
+    options.setRunner(DataflowRunner.class);
+    options.setJobName(workloadJobName);
+
+    final PipelineResult pipelineResult = WorkloadGenerator.generateWorkload(options);
+
+    MetricServiceClient metricServiceClient = MetricServiceClient.create();
+    ProjectName name = ProjectName.of(projectId);
+
+    // Wait X minutes and then get metrics for the X minute period.
+    Thread.sleep(WAIT_DURATION);
+    long startMillis = System.currentTimeMillis() - WAIT_DURATION;
+
+    TimeInterval interval =
+        TimeInterval.newBuilder()
+            .setStartTime(Timestamps.fromMillis(startMillis))
+            .setEndTime(Timestamps.fromMillis(System.currentTimeMillis()))
+            .build();
+
+    ListTimeSeriesRequest request =
+        ListTimeSeriesRequest.newBuilder()
+            .setName(name.toString())
+            .setFilter("metric.type=\"bigtable.googleapis.com/server/request_count\"")
+            .setInterval(interval)
+            .build();
+    ListTimeSeriesPagedResponse response = metricServiceClient.listTimeSeries(request);
+
+    long startRequestCount = 0;
+    long endRequestCount = 0;
+    for (TimeSeries ts : response.iterateAll()) {
+      startRequestCount = ts.getPoints(0).getValue().getInt64Value();
+      endRequestCount = ts.getPoints(ts.getPointsCount() - 1).getValue().getInt64Value();
+    }
+    assertThat(endRequestCount - startRequestCount > rate);
+
+    // Stop the running job.
+    String jobId = ((DataflowPipelineJob) pipelineResult).getJobId();
```

Comment:
should this be in a try/catch/finally so we always stop the job? or would the job stop itself if there are errors?

Reply:
if there is an issue the job would probably fail to build